libs/flexIPC/fleXdIPCCommon/src/FleXdIPCMsg.cpp

```diff
@@ -0,0 +1,466 @@
+            {
+                return m_next;
+            }
+            FleXdIPCAdtHdr* FleXdIPCAdtHdr::getNext()
+            {
+                return m_next;
+            }
+
+            ////////////////
+            // FleXdIPCMsg
+            ////////////////
+
+            FleXdIPCMsg::FleXdIPCMsg(uint8_t msgType, std::vector<uint8_t>&& payload)
+            : m_payload(payload),
+              m_headerParamType(0),
+              m_headerParam(msgType),
+              m_additionalHeader(nullptr),
+              m_valid(true)
+            {
+            }
+
+            FleXdIPCMsg::FleXdIPCMsg(bool headerParamType, uint8_t headerParam, std::vector<uint8_t>&& payload)
+            : m_payload(payload),
+              m_headerParamType(headerParamType ? IPC_MSG_HEADER_PARAM_TYPE_ADDITIONAL_HEADER : IPC_MSG_HEADER_PARAM_TYPE_MSG_TYPE),
+              m_headerParam(headerParamType ? IPC_MSG_HEADER_IN_PAYLOAD_FLAG : headerParam),
+              m_additionalHeader(nullptr),
+              m_valid(true)
+            {
+            }
+
+            FleXdIPCMsg::FleXdIPCMsg(FleXdIPCAdtHdr*& additionalHeader, std::vector<uint8_t>&& payload)
+            : m_payload(payload),
+              m_headerParamType(IPC_MSG_HEADER_PARAM_TYPE_ADDITIONAL_HEADER),
+              m_headerParam(additionalHeader != nullptr ? additionalHeader->getMask() : 0),
+              m_additionalHeader(additionalHeader),
+              m_valid(true)
+            {
+                additionalHeader = nullptr;
+            }
+
+            FleXdIPCMsg::FleXdIPCMsg(std::vector<uint8_t>&& payload, bool calcCRC /*= true*/)
+            : m_payload(payload),
+              m_headerParamType(1),
+              m_headerParam(0),
+              m_additionalHeader(calcCRC ? new FleXdIPCAdtHdr() : nullptr),
+              m_valid(true)
+            {
+                if (calcCRC) m_additionalHeader->setValue_2(calcCRC16(&m_payload[0] , m_payload.size()));
+            }
+
+            FleXdIPCMsg::~FleXdIPCMsg()
+            {
+                if (m_additionalHeader != nullptr)
+                {
+                    delete m_additionalHeader;
+                }
+            }
+
+            FleXdIPCMsg::FleXdIPCMsg(FleXdIPCMsg&& other)
+            : m_payload(std::move(other.m_payload)),
+              m_headerParamType(other.m_headerParamType),
+              m_headerParam(other.m_headerParam),
+              m_additionalHeader(other.m_additionalHeader),
+              m_valid(other.m_valid)
+            {
+                other.m_headerParamType = 0;
+                other.m_headerParam = 0;
+                other.m_additionalHeader = nullptr;
+                other.m_valid = false;
+            }
+
+            FleXdIPCMsg& FleXdIPCMsg::operator=(FleXdIPCMsg&& other)
+            {
+                if (this != &other)
+                {
+                    m_payload.clear();
+                    if (m_additionalHeader != nullptr) delete m_additionalHeader;
```

Comment:
Ditto 3. 'if (m_next != nullptr)' is fairly useless since standard defines the behaviour of 'delete' well when 'nullptr' is supplied.